### PR TITLE
[HOPSWORKS-541] Export/create Anaconda environment from .yml

### PR DESCRIPTION
**Pull request by robzor92**
## Make sure there is no duplicate PR for this issue

* **Please check if the PR meets the following requirements**
- [x] Adds tests for the submitted changes (for bug fixes & features)
- [ ] Passes the tests
- [x] HOPSWORKS JIRA issue has been opened for this PR
- [x] All commits have been squashed down to a single commit

* **Post a link to the associated JIRA issue**

* **What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)

* **What is the new behavior (if this is a feature change)?**

* **Does this PR introduce a breaking change?** (What changes might users need to make in their application due to this PR?)

* **Other information**:

### CODE REVIEW

**Comment by robzor92** [comment]
https://github.com/hopshadoop/hops-testing/pull/93

File: hopsworks-api/src/main/java/io/hops/hopsworks/api/pythonDeps/PythonDepsService.java [hunk 1]
```diff
@@ -348,6 +464,28 @@ public Response removeEnv(@Context SecurityContext sc, @Context HttpServletReque
...
+    String cpuHost = hostsFacade.findCPUHost();
+    if (cpuHost != null) {
+      exportEnvironment(cpuHost, "environment_cpu.yml", hdfsUser);
+    }
+
+    String gpuHost = hostsFacade.findGPUHost();
+    if (gpuHost != null) {
+      exportEnvironment(gpuHost, "environment_gpu.yml", hdfsUser);
+    }
+
```
Comment (jimdowling) [inline hopsworks-api/src/main/java/io/hops/hopsworks/api/pythonDeps/PythonDepsService.java]: if both cpuHost and gpuHost are null, there will not be an error.
if (cpuHost ==null && gpuHost == null) {
   throw new AppExeception(.... "Export environment failed. Could not find any hosts.");
}

File: hopsworks-api/src/main/java/io/hops/hopsworks/api/pythonDeps/PythonDepsService.java [hunk 2]
```diff
@@ -619,4 +757,68 @@ public Response search(@Context SecurityContext sc,
...
   }
+
+  private String getYmlFromPath(String path, String username) throws AppException {
+
+    DsPath ymlPath = pathValidator.validatePath(this.project, path);
+    ymlPath.validatePathExists(inodes,false);
+    org.apache.hadoop.fs.Path fullPath = ymlPath.getFullPath();
+    String ymlFileName = fullPath.getName();
+
+    DistributedFileSystemOps udfso = null;
```
Comment (jimdowling) [inline hopsworks-api/src/main/java/io/hops/hopsworks/api/pythonDeps/PythonDepsService.java]: Close this.
See ProjectService.java - line 509.

File: hopsworks-api/src/main/java/io/hops/hopsworks/api/pythonDeps/PythonDepsService.java [hunk 3]
```diff
@@ -619,4 +757,68 @@ public Response search(@Context SecurityContext sc,
...
+    FSDataInputStream is = null;
+
+    try {
+      udfso = dfs.getDfsOps(username);
```
Comment (jimdowling) [inline hopsworks-api/src/main/java/io/hops/hopsworks/api/pythonDeps/PythonDepsService.java]: same with dfs, needs to be closed.

File: hopsworks-common/src/main/java/io/hops/hopsworks/common/dao/host/HostsFacade.java [hunk 4]
```diff
@@ -86,6 +86,27 @@ public Hosts findByHostname(String hostname) {
     return query.getResultList();
   }
 
+  public String findCPUHost() {
+    List<Hosts> hosts = find();
+    for(Hosts host: hosts){
+      if(host.getNumGpus() == 0){
```
Comment (jimdowling) [inline hopsworks-common/src/main/java/io/hops/hopsworks/common/dao/host/HostsFacade.java]: Check
if (host.getNumGpus() ==0 && host.isAnacondaEnabled())

File: hopsworks-common/src/main/java/io/hops/hopsworks/common/dao/host/HostsFacade.java [hunk 5]
```diff
@@ -86,6 +86,27 @@ public Hosts findByHostname(String hostname) {
...
+        return host.getHostname();
+      }
+    }
+    return null;
+  }
+
+  public String findGPUHost() {
+    List<Hosts> hosts = find();
+    for(Hosts host: hosts){
+      if(host.getNumGpus() > 0){
```
Comment (jimdowling) [inline hopsworks-common/src/main/java/io/hops/hopsworks/common/dao/host/HostsFacade.java]: Check
if (host.getNumGpus() >0 && host.isAnacondaEnabled())

**Pull request merged**